js/indicators/ichimoku-kinko-hyo.src.js

```diff
@@ -396,50 +485,203 @@ seriesType('ikh', 'sma',
+                    // check the middle point (if exist)
+                    if (Math.floor(sectionPoints.length / 2) >= 1) {
+                        var x = Math.floor(sectionPoints.length / 2);
+
+                        // when middle points has equal values
+                        // compare all ponints plotY value sum
+                        if (
+                            sectionPoints[x].plotY ===
+                            sectionNextPoints[x].plotY
+                        ) {
+
+                            pointsPlotYSum = 0;
+                            nextPointsPlotYSum = 0;
+
+                            for (k = 0; k < sectionPoints.length; k++) {
+                                pointsPlotYSum +=
+                                sectionPoints[k].plotY;
+                                nextPointsPlotYSum +=
+                                sectionNextPoints[k].plotY;
+                            }
+
+                            if (pointsPlotYSum > nextPointsPlotYSum) {
+                                points[0] = points[0].concat(sectionPoints);
+                                nextPoints[0] =
+                                    nextPoints[0].concat(sectionNextPoints);
+                            } else {
+                                points[1] = points[1].concat(sectionPoints);
+                                nextPoints[1] =
+                                    nextPoints[1].concat(sectionNextPoints);
+                            }
+
+                        } else {
+                            // compare middle point of the section
+                            if (
+                                sectionPoints[x].plotY >
+                                sectionNextPoints[x].plotY
+                            ) {
+                                points[0] = points[0].concat(sectionPoints);
+                                nextPoints[0] =
+                                    nextPoints[0].concat(sectionNextPoints);
+                            } else {
+                                points[1] = points[1].concat(sectionPoints);
+                                nextPoints[1] =
+                                    nextPoints[1].concat(sectionNextPoints);
+                            }
+                        }
+                    } else {
+                        // compare first point of the section
+                        if (sectionPoints[0].plotY >
+                            sectionNextPoints[0].plotY
+                        ) {
+                            points[0] = points[0].concat(sectionPoints);
+                            nextPoints[0] =
+                                nextPoints[0].concat(sectionNextPoints);
+                        } else {
+                            points[1] = points[1].concat(sectionPoints);
+                            nextPoints[1] =
+                                nextPoints[1].concat(sectionNextPoints);
+                        }
```

Comment:
Lines 628 - 638 are the same like 613 - 623 (except simple IF of cors). 601-608 also.

 Consider optimization of them?